cliquet/authorization.py

```diff
@@ -0,0 +1,74 @@
+import re
+import six
+from pyramid.security import IAuthorizationPolicy
+from zope.interface import implementer
+
+from cliquet import utils
+from cliquet.storage import exceptions as storage_exceptions
+
+
+@implementer(IAuthorizationPolicy)
+class AuthorizationPolicy(object):
+    def permits(self, context, principals, permission):
+        if permission == 'dynamic':
+            permission = context.required_permission
+        object_id = context.object_id
+        get_bound_permissions = context.get_bound_permissions
+        has_permission = context.has_permission
+
+        if permission == 'create':
+            permission = '%s:%s' % (context.object_type, permission)
+        return has_permission(object_id, permission, principals,
+                              get_bound_permissions)
```

Comment:
Here, only ``permission`` and ``principals`` are coming from this function.
Is context supposed to be a stateless struct or could we curry ``has_permission()`` with ``object_id and get_bound_permission ?